### PR TITLE
Fix filtering

### PR DESCRIPTION
**Pull request by jasononaquest**
- Get multi-word work and venue filters to work (need to mark field names as `facetable`)
- Get multi-term venue search to work (resurrect and refactor the `add_facet_fq_to_solr` override)
- Support the `Other` venue search by doing a negated search
- Clear out production credit/year fields when no id/date_created provided

### CODE REVIEW

File: app/models/file_search.rb [hunk 1]
```diff
@@ -80,16 +80,17 @@ def files
     params.empty? ? highlighted_files : filtered_files
   end
 
-  private
-
-  VENUES = ["Elizabethan", "Angus Bowmer", "Thomas", "The Green Show"]
+  PRIMARY_VENUES = ["Elizabethan", "Angus Bowmer", "Thomas", "The Green Show"]
```
Comment (seanculver) [inline app/models/file_search.rb:83]: Have you considered putting these constants at the top of the file?

Comment (randycoulman) [inline app/models/file_search.rb:83]: They are essentially private, but used by the tests, so we put them here on purpose.

File: app/models/search_builder.rb [hunk 2]
```diff
@@ -3,4 +3,46 @@ class SearchBuilder < Blacklight::SearchBuilder
...
+      .map { |value| %Q[#{field}:"#{value}"] }
+      .join(conjuction(field))
+  end
+
+  def single_search_term(field, value)
+    facet_value_to_fq_string(field, value)
+  end
+
+  def conjuction(field)
+    field[0] == "!" ? " AND " : " OR "
```
Comment (seanculver) [inline app/models/search_builder.rb]: Could this be `start_with?`? ;)

**Comment by seanculver** [comment]
:+1:

**Pull request merged**